packages/stream_chat_flutter/lib/src/scroll_view/channel_scroll_view/stream_channel_list_tile.dart

```diff
@@ -216,26 +216,40 @@ class StreamChannelListTile extends StatelessWidget {
                 initialData: channelState.messages,
                 comparator: const ListEquality().equals,
                 builder: (context, messages) {
-                  final lastMessage = messages.lastWhereOrNull(
-                    (m) => !m.shadowed && !m.isDeleted,
-                  );
-
-                  if (lastMessage == null ||
-                      (lastMessage.user?.id != currentUser.id)) {
-                    return const Offstage();
-                  }
-
-                  return Padding(
-                    padding: const EdgeInsets.only(right: 4),
-                    child:
-                        sendingIndicatorBuilder?.call(context, lastMessage) ??
+                  return BetterStreamBuilder<List<Read>>(
+                    stream: channelState.readStream,
+                    initialData: channelState.read,
+                    builder: (context, reads) {
+                      final lastMessage = messages.lastWhereOrNull(
```

Comment:
With the new `readStream` we can remove the `messagesStreamBuilder` and rely on the `channelState.messages`.
```
final lastMessage = channelState.messages.lastWhereOrNull(
```